tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/domainmodels/codegeneration/ServiceModel.java

```diff
@@ -44,4 +44,25 @@ public boolean hasStreamingRequestShapes() {
     public Collection<Error> getNonCoreServiceErrors() {
         return serviceErrors.stream().filter(e -> !e.isCoreError()).collect(Collectors.toSet());
     }
+
+    public boolean hasSigV4Auth() {
+        if(metadata.getSignatureVersion().equals("v4") || metadata.getSignatureVersion().equals("s3v4")) {
+            return true;
+        }
+        return operations.values().parallelStream().anyMatch(operation -> operation.getSignerName().equals("Aws::Auth::SIGV4_SIGNER"));
```

Comment:
parallel steams are generally advised against in the java world because they use the common fork-join pool, meaning they share an executor pool with the rest of the world. I would either use sequential streams which is often faster. or alternatively it can have its own executor pool.